jvm/src/main/groovy/org/asciidoctor/gradle/jvm/AbstractAsciidoctorTask.groovy

```diff
@@ -594,6 +594,19 @@ class AbstractAsciidoctorTask extends AbstractAsciidoctorBaseTask {
         executorConfigurations
     }
 
+    /** The main property will be removed in Gradle 8.0 and replaced by the mainClass property.
+     *
+     * Attempt to use the main property, and if the method doesn't exist try the replacement.
+     */
+    @CompileDynamic
+    private void setExecClass(JavaExecSpec jes, String execClass) {
```

Comment:
I think you should check the Gradle version, and if the version already has the new property then use it, and if it doesn't call the old method via reflection. This avoids the overhead of throwing an exception.

Reply:
Probably better. I think mainClass was added in either 6.4 or 6.6.

Reply:
Good idea, I updated the PR.